cmd/ipfs/daemon.go

```diff
@@ -452,8 +429,8 @@ func serveHTTPGateway(req cmds.Request) (error, <-chan error) {
 	if err != nil {
 		return fmt.Errorf("serveHTTPGateway: req.Option(%s) failed: %s", writableKwd, err), nil
 	}
-	if !writableOptionFound {
-		writable = cfg.Gateway.Writable
+	if writableOptionFound {
+		cfg.Gateway.Writable = writable
```

Comment:
So... the second return value (writableOptionFound) is separate from defaults -- it tells specifically whether the option was passed on the CLI.

So this assumption is still true and everything's good as-is:

> From what I can tell, --writable if present is supposed to override cfg.Gateway.Writable.

In other words, the default value for --writable is never used -- only if it has actually been passed.